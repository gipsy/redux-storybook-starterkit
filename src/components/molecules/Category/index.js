import React from 'react'
import PropTypes from 'prop-types'
import styled, { css } from 'styled-components'
import breakpoint from 'styled-components-breakpoint'
import { font, palette } from 'styled-theme'
import { ifProp } from 'styled-tools'
import { placeholder, rem, transitions } from 'polished'
import Icon from '~/src/components/atoms/Icon'
import Link from '~/src/components/atoms/Link'

import { responsiveFontSizes as fontSize } from '~/src/components/themes/default'

const styles = css`
  display: inline-block;
  position: relative;
  font-family: ${font('primary')};
  font-weight: 600;
  text-align: center;
  box-sizing: border-box;
  background-color: ${palette('grayscale', 0, true)};
  cursor: pointer;
  ${transitions('border-color .1s ease-in')};

  &:hover {
    border-color: ${palette('grayscale', 3, true)}
  }

  @media screen and (max-width: 420px) {
    font-size: ${fontSize.mobile};
    border: 2px solid ${palette('grayscale', 2, true)};
    border-radius: 6px;
    height: 210px;
    width: ${({ responsive }) => responsive ? '100%' : rem('135px')};
  }

  @media screen and (min-width: 421px) {
    font-size: ${fontSize.desktop};
    border: 3px solid ${palette('grayscale', 2, true)};
    border-radius: 6px;
    height: 245px;
    width: ${({ responsive }) => responsive ? '100%' : rem('160px')};
  }
`

const StyledCategory = styled.div`${styles}`

const StyledIcon = styled(Icon)`
  color: ${palette('black', 0)};

  @media screen and (max-width: 420px) {
    font-size: ${rem('46.6px')};
    margin: ${({ responsive }) => responsive && 0};
    margin-top: ${rem('60px')};
  }

  @media screen and (min-width: 421px) {
    font-size: ${rem('66px')};
    margin-top: ${rem('60px')};
  }
`

const Text = styled.span`
  font-family: ${font('primary')};
  color: ${palette('grayscale', 0)};
  position: absolute;
  left: 0;
  right: 0;
  text-transform: uppercase;

  @media screen and (max-width: 420px) {
    font-size: ${fontSize.mobile};
    line-height: ${rem('16px')};
    font-weight: 500;
    margin: 0 ${rem('20px')};
    bottom: ${rem('19px')};
  }

  @media screen and (min-width: 421px) {
    font-size: ${fontSize.desktop};
    line-height: ${rem('20px')};
    font-weight: 600;
    margin: 0 ${rem('30px')};
    bottom: ${rem('27px')};
  }
`

const Category = ({ width, height, icon, responsive, children, ...props }) => {
  const iconElement = (
    <StyledIcon
      icon={icon}
      responsive={responsive}
      palette={props.palette}
      reverse={props.reverse}
      width={width}
      height={height}
    />
  )
  return (
    <Link {...props}>
      <StyledCategory>
        { iconElement }
        <Text>{children}</Text>
      </StyledCategory>
    </Link>
  )
}

Category.propTypes = {
  icon: PropTypes.string,
  children: PropTypes.string,
  reverse: PropTypes.bool,
  children: PropTypes.node,
  width: PropTypes.integer,
  height: PropTypes.integer,
}

export default Category
