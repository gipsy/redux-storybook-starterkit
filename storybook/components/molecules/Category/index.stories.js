import React from 'react'
import { storiesOf } from '@storybook/react'
import Category from '~/src/components/molecules/Category'

storiesOf('Components/Molecules', module)
  .addWithChapters(
    'Category' ,
    {
      subtitle: 'Category Description',
      info: `
        The front end team have worked to abstract the view layer out of Rails over to React enabling us to be much more component oriented in the code. At the same time, the design team have moved to Sketch and have started to build out a pattern library that will be updated and maintained directly alongside the living components.
      `,
      chapters: [
        // List of chapters
        {
          title: 'Category Component',
          subtitle: 'New Category Component',
          info: `
            Our primary Category is transparent.
          `,
          sections: [
            // List of sections
            {
              title: 'This is default Category component',
              subtitle: 'Use this component in your project',
              sectionFn: () => (
                <Category
                  icon="code"
                  responsive
                  width={46}
                  height={26}
                  onClick={() => { alert('Hello World!'); }}
                >
                  tech / project
                </Category>
              ),
              options: {
                showSource: true,
                allowSourceToggling: true,
                showPropTables: false,
                allowPropTablesToggling: true,
              },
            },
          ]
        }
      ]
    }
  )
