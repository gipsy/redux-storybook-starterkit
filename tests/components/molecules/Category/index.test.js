import React from 'react'
import { shallow } from 'enzyme'
import Category from '~/src/components/molecules/Category'

const wrap = (props = {}) => shallow(<Category {...props} />)

it('renders children when passed in', () => {
  const wrapper = wrap({ children: 'test' })
  expect(wrapper.contains('test')).toBe(true)
})

it('renders props when passed in', () => {
  const wrapper = wrap({ id: 'foo' })
  expect(wrapper.find({ id: 'foo' })).toHaveLength(1)
})

test('<Category /> renders with all allowed props', () => {
  const wrapper = wrap({
    id: 'test',
    children: 'test',
    palette: 'primary',
    reverse: true,
    responsive: true,
    width: 46,
    height: 26,
    icon: 'path/to/icon'
  })
  expect(wrapper).toMatchSnapshot()
})
