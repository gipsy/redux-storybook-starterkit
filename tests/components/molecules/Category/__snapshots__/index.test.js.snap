// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Category /> renders with all allowed props 1`] = `
<Link
  id="test"
  palette="primary"
  reverse={true}
>
  <styled.div>
    <Styled(Icon)
      height={26}
      icon="path/to/icon"
      palette="primary"
      responsive={true}
      reverse={true}
      width={46}
    />
    <styled.span>
      test
    </styled.span>
  </styled.div>
</Link>
`;
